src/entity2/revision.ts

```diff
@@ -0,0 +1,55 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, JoinColumn, ManyToOne } from 'typeorm';
+
+// eslint-disable-next-line import/no-cycle
+import { Dataset } from './dataset';
+import { User } from './user';
+
+interface Revision {
+    id: string;
+    revision_index: number;
+    dataset: Dataset;
+    creation_date: Date;
+    previous_revision: Revision;
+    online_cube_filename: string;
+    publish_date: Date;
+    approval_date: Date;
+    approved_by: User;
+    created_by: User;
+}
+
+@Entity()
+export class RevisionEntity extends BaseEntity implements Revision {
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
+
+    @Column({ type: 'int' })
+    revision_index: number;
+
+    @ManyToOne(() => Dataset, (dataset) => dataset.revisions)
+    @JoinColumn({ name: 'dataset_id' })
+    dataset: Dataset;
+
+    @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
+    creation_date: Date;
+
+    @ManyToOne(() => RevisionEntity, { nullable: true })
```

Comment:
There's only one "previous" revision for each revision but there could be zero, one or many "next" revisions for a particular revision.

Reply:
So I am treating them almost like a linked listed...  The latest revision points to the previous revision and that revision then points to the previous revision, because Revisions can be abandoned you don't get a one to one relationship between revisions so ManyToOne here is correct.  It can also be null as there might not be a previous revision as is the case for the first revision created.